## Задание 2

###  В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
### Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
### оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
### значение с доверительной вероятностью 0,95.

#### Поскольку неизвестно среднее квадратичное генеральной совокупности, то используем  критерий Стьюдента:  распределение  Стьюдента зависит от степеней свободы (n - 1).  α/2  = 0.025
#### По таблице  берем параметры 0.975 для p и 9 степеней свободы. Получаем t=2.262


In [1]:
import numpy as np
from scipy import stats
x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
x1 = np.mean(x)
x1

6.590000000000001

In [3]:
D1 = np.var(x, ddof=1)
D1

0.2032222222222223

In [4]:
t1 = 2.262
x1 - t1*np.sqrt(D1/10)

6.267538255912426

In [5]:
x1 + t1*np.sqrt(D1/10)

6.912461744087575

Ответ: (6.27, 6.91)

# Задание 3

###  Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170
### Рост матерей 178, 165, 165, 173, 168, 155, 160, 164, 178, 175

#### Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

In [6]:
import numpy as np
from scipy import stats
x = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
y = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])
x1 = np.mean(x)
x1

166.2

In [7]:
len(x)

10

In [8]:
len(y)

10

In [9]:
y1 = np.mean(y)
y1

168.1

In [10]:
delta = y1 - x1
delta

1.9000000000000057

In [11]:
D1 = np.var(x, ddof=1)
D1

91.06666666666666

In [12]:
D2 = np.var(y, ddof=1)
D2

60.10000000000001

In [14]:
D = (D1 + D2)/2
D

75.58333333333334

In [15]:
SE = np.sqrt(D/10 + D/10)
SE

3.8880157750022915

In [16]:
t = stats.t.ppf(0.975, 18)
t

2.10092204024096

In [17]:
delta - t*SE

-6.268418034506846

In [18]:
delta + t*SE

10.068418034506857

Ответ: (-6.27, 10.07) 